services/web/server/tests/unit/with_dbs/10/meta/test_meta_iterations.py

```diff
@@ -0,0 +1,232 @@
+    "exclude_none": True,  # e.g. thumbnail: None will fail validation
+    "exclude_unset": True,
+}
+
+# FIXTURES ----------------------------------
+
+
+@pytest.fixture
+async def context_with_logged_user(client: TestClient, logged_user: AUserDict):
+
+    yield
+
+    assert client.app
+    engine = client.app[APP_DB_ENGINE_KEY]
+    async with engine.acquire() as conn:
+        # cascade deletes everything except projects_vc_snapshot
+        await conn.execute(projects.delete())
+
+
+# TESTS ----------------------------------
+
+
+@pytest.mark.acceptance_test
+async def test_iterators_workflow(
+    client: TestClient, context_with_logged_user: None, mocker, faker: Faker
+):
+    resp: ClientResponse
+
+    # check init meta is correct
+    assert client.app
+    assert projects_redirection_middleware in client.app.middlewares
+    assert get_project_run_policy(client.app) == meta_project_policy
+
+    # NEW project --------------------------------------------------------------
+    mocker.patch(
+        "simcore_service_webserver.director_v2_api.create_or_update_pipeline",
+        return_value=None,
+    )
+    mocker.patch(
+        "simcore_service_webserver.director_v2_api.get_computation_task",
+        return_value=None,
+    )
+    # ----
+
+    assert NEW_PROJECT.request_desc == "POST /v0/projects"
+    resp = await client.post("/v0/projects", json=NEW_PROJECT.request_payload)
+    assert resp.status == NEW_PROJECT.status_code, await resp.text()
+    body = await resp.json()
+
+    project_data: ProjectDict = body["data"]
+    project_uuid = project_data["uuid"]
+
+    # CREATE meta-project: iterator 0:3 -> sleeper -> sleeper_2 ---------------
+    modifications = REPLACE_PROJECT_ON_MODIFIED.request_payload
+    assert modifications
+    project_data.update({key: modifications[key] for key in ("workbench", "ui")})
+    project_data["ui"].setdefault("currentNodeId", project_uuid)
+
+    resp = await client.put(
+        f"/v0/projects/{project_data['uuid']}",
+        json=project_data,
+    )
+    assert resp.status == REPLACE_PROJECT_ON_MODIFIED.status_code, await resp.text()
+
+    # TODO: create iterations, so user could explore parametrizations?
+
+    # RUN metaproject ----------------------------------------------------------
+    async def _mock_start(project_id, user_id, **options):
+        return f"{project_id}"
+
+    mocker.patch(
+        "simcore_service_webserver.director_v2_core.DirectorV2ApiClient.start",
+        side_effect=_mock_start,
+    )
+    # ----
+
+    resp = await client.post(
+        f"/v0/computation/pipeline/{project_uuid}:start",
+        json=RUN_PROJECT.request_payload,
+    )
+    data, _ = await assert_status(resp, web.HTTPCreated)
+    assert project_uuid == data["pipeline_id"]
+    ref_ids = data["ref_ids"]
+    assert len(ref_ids) == 3
+
+    # TODO: check: has auto-commited
+    # TODO: check: has iterations as branches
+    # TODO: retrieve results of iter1
```

Comment:
still relevant?

Reply:
all TODOs in this tests are reminders for features that will be added in next PR